File: src/ui/AutoPodcaster/Models/Input.cs
namespace AutoPodcaster.Models;

public sealed class Input
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string? Title { get; set; }
    public string? Date { get; set; }
    public string? LastUpdated { get; set; }
    public string? Status { get; set; }
    public string? Author { get; set; }
    public string? Description { get; set; }
    public string? Source { get; set; }
    public string? Type { get; set; }
    public string? ThumbnailUrl { get; set; }
    public List<string>? Topics { get; set; }
    public List<string>? Entities { get; set; }
    public string? Content { get; set; }
}

File: src/autopodcaster_model/autopodcaster_model/Input.py
class Input:
    id: str
    title: str
    date: str
    last_updated: str
    status: str
    author: str = ''
    description: str = ''
    source: str
    type: str
    thumbnail_url: str = ''
    topics: list = []
    entities: list = []
    content: str = ''

    def to_dict(self):
        return {
            "id": self.id,
            "title": self.title,
            "date": self.date,
            "last_updated": self.last_updated,
            "status": self.status,
            "author": self.author,
            "description": self.description,
            "source": self.source,
            "type": self.type,
            "thumbnail_url": self.thumbnail_url,
            "topics": self.topics,
            "entities": self.entities,
            "content": self.content
        }
